tools/pythonpkg/duckdb-stubs/__init__.pyi

```diff
@@ -412,7 +412,7 @@ class DuckDBPyRelation:
     def list(self, column: str, groups: str = ..., window_spec: str = ..., projected_columns: str = ...) -> DuckDBPyRelation: ...
 
     def arrow(self, batch_size: int = ...) -> pyarrow.lib.Table: ...
-    def __arrow_c_stream__(self) -> object: ...
+    def __arrow_c_stream__(self, requested_schema: Any) -> object: ...
```

Comment:
I'm not sure if this will produce any issues, but the spec document suggests typing this as `object` https://arrow.apache.org/docs/format/CDataInterface/PyCapsuleInterface.html#protocol-typehints

Reply:
Actually, this will mess with type checkers as long as you don't have `=None`

Pylance doesn't allow this:

```py

from typing import Protocol, Any


class ArrowStreamExportable(Protocol):
    def __arrow_c_stream__(self, requested_schema: object | None = None) -> object: ...


def accepts_arrow(data: ArrowStreamExportable): ...


class ArrowObject:
    def __arrow_c_stream__(self, requested_schema: Any) -> object: ...

obj = ArrowObject()
accepts_arrow(obj)
```

with error:

```
Argument of type "ArrowObject" cannot be assigned to parameter "data" of type "ArrowStreamExportable" in function "accepts_arrow"
  "ArrowObject" is incompatible with protocol "ArrowStreamExportable"
    "__arrow_c_stream__" is an incompatible type
      Type "(requested_schema: Any) -> object" is not assignable to type "(requested_schema: object | None = None) -> object"
        Parameter "requested_schema" is missing default argument
```

Reply:
Nice catch - thanks!

Reply:
Thanks!